### PR TITLE
LPS-73997 Unable to run Search unit tests from Eclipse due to PropsValues reference

### PR DESCRIPTION
**Pull request by arboliveira**
Hi Shuyang, I thought I should run this by you first -- hoping you can help check I'm not introducing any instability. Did my best to only make changes that would keep integration tests running exactly the same as before, and fixed a couple of unit test assertions that were flaky and became more evident after my changes.

https://issues.liferay.com/browse/LPS-73997

### CODE REVIEW

File: modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java [hunk 1]
```diff
@@ -143,15 +135,39 @@ private void _assertPagination(int start, int end, int expectedSize)
 		searchContext.setEnd(end);
 		searchContext.setStart(start);
 
-		Hits hits = indexSearcher.search(searchContext, getDefaultQuery());
+		IdempotentRetryAssert.retryAssert(
```
Comment (shuyangzhou) [inline modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java]: Can we avoid using IdempotentRetryAssert? I really don't like this, by using this it means we have random behavior. There is no way to know 3sec(or any other time) is enough.

**Comment by shuyangzhou** [comment]
See comment above, and we should try to remove IdempotentRetryAssert completely if we can.

File: modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java (hunk 1 above)
Comment (arboliveira) [inline modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java]: Unfortunately we can't avoid IdempotentRetryAssert here. I know you don't like it, but this is not a Portal integration test - there's no `@Sync` to use, and it's uniform with dozens of other `portal-search-elasticsearch` unit tests using it already. If you want, we can discuss in-depth approaches to remove IdempotentRetryAssert from unit tests, as a task for a separate ticket - but this bug is orthogonal to that, and I'd rather not block fixing it (it affects the entire Search team) unless you already have a solution ready to apply.

Comment (shuyangzhou) [inline modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java]: Ok, I thought you were asking for checking for instability, which IdempotentRetryAssert is one. If you are not planning to change that, then this pull is fine to me, it does not really change anything for real. Feel free to proceed on this.

Comment (arboliveira) [inline modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java]: Thanks Shuyang, I really appreciate it. I will prioritize a technical debt issue to remove IdempotentRetryAssert completely.

Comment (shuyangzhou) [inline modules/apps/foundation/portal-search/portal-search-test-util/src/main/java/com/liferay/portal/search/test/util/indexing/BaseIndexSearcherTestCase.java]: Thanks @arboliveira ! Let me know once you have it done, I'd like to see it goes away :)

**Pull request closed without merging**